daemon.go

```diff
@@ -201,3 +207,23 @@ func (d *Daemon) Close() error {
 
 	return merr.ErrorOrNil()
 }
+
+func (d *Daemon) awaitTermination() {
+	go func() {
```

Comment:
The name `awaitSomething()` usually implies that the function is blocking.

One option is to rename it, however I think it's better to indeed make it blocking (instead of having a function that just launches a goroutine and exits, which is a bit misleading). In this case, we can call it as `d.terminateOnce.Do(func () { go d.awaitTermination() })`.